core/src/main/kotlin/fr/sncf/osrd/graph/PathfindingGraph.kt

```diff
@@ -0,0 +1,60 @@
+package fr.sncf.osrd.graph
+
+import fr.sncf.osrd.sim_infra.api.Block
+import fr.sncf.osrd.stdcm.graph.extendLookaheadUntil
+import fr.sncf.osrd.stdcm.infra_exploration.InfraExplorer
+import fr.sncf.osrd.utils.units.Length
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+
+data class PathfindingNode(
+    val infraExplorer: InfraExplorer,
+    val previousEdge: PathfindingEdge,
+    val locationOnEdge: Offset<Block>?,
+)
+
+data class PathfindingEdge(
```

Comment:
Needs a good `.equals` and `.hash`, we don't actually mark anything as visited there

Reply:
See https://github.com/OpenRailAssociation/osrd/pull/9153/commits/f62e41b838b6b83281149893d500d2f2796da6d6